python/ID_EX.py

```diff
@@ -1,67 +1,55 @@
 from myhdl import *
 
 @block
-def id_ex(clk, readA_in, readB_in, offset_in, PC_in, RTa_in,
-        RTb_in, RS_in, RD_in, WB_in, MemToReg_in, MemWrite_in,
-        MemRead_in, lwFlush_in, brFlush_in, funct_in, opCode_in,
-        readA_out, readB_out, offset_out, PC_out, RTa_out,
-        RTb_out, RS_out, RD_out, WB_out, MemToReg_out, MemWrite_out,
-        MemRead_out, lwFlush_out, brFlush_out, funct_out, opCode_out):
-
-    class latches:
-        readA = intbv(0, 0, 2**32)
-        readB = intbv(0, 0, 2**32)
-        offset = intbv(0, 0, 2**32)
-        PC = intbv(0, 0, 2**32)
-        RTa = intbv(0, 0, 2**5)
-        RTb = intbv(0, 0, 2**5)
-        RS = intbv(0, 0, 2**5)
-        RD = intbv(0, 0, 2**5)
-        WB = intbv(0, 0, 2**1)
-        MemToReg = intbv(0, 0, 2**1)
-        MemWrite = intbv(0, 0, 2**1)
-        MemRead = intbv(0, 0, 2**1)
-        lwFlush = intbv(0, 0, 2**1)
-        brFlush = intbv(0, 0, 2**1)
-        funct = intbv(0, 0, 2**6)
-        opCode = intbv(0, 0, 2**3)
+def id_ex(clk, rst, ID_Hazard_lwstall, ID_Hazard_Branch, Branch_in, MemRead_in,
+            MemWrite_in, Jump_in, RegWrite_in, ALUSrc_in, ALUOp_in,
+            RegDst_in, MemtoReg_in, jump_addr_in, PC_plus4_in,
```

Comment:
It looks like branch_addr_in is missing. https://github.com/txstate-pcarch-blue/CPU/blob/master/verilog/ID_EX.v#L38